internal/app/routes/route.go

```diff
@@ -4,15 +4,27 @@ import (
 	"net/http"
 
 	"github.com/go-chi/chi/v5"
+	"go.uber.org/zap"
 
 	"github.com/Erlast/short-url.git/internal/app/config"
 	"github.com/Erlast/short-url.git/internal/app/handlers"
+	"github.com/Erlast/short-url.git/internal/app/middlewares"
 	"github.com/Erlast/short-url.git/internal/app/storages"
 )
 
-func NewRouter(store *storages.Storage, conf *config.Cfg) *chi.Mux {
+func NewRouter(store storages.URLStorage, conf *config.Cfg, logger *zap.SugaredLogger) *chi.Mux {
 	r := chi.NewRouter()
 
+	WithLogging := func(h http.Handler) http.Handler {
```

Comment:
1. Локальные переменные в Go не называются с заглавной буквы
2. Лучше вообще не создавать здесь дополнительную переменную:

```suggestion
        r.Use(func(h http.Handler) http.Handler {
```